### PR TITLE
authenticated message subscription endpoints

### PR DESCRIPTION
**Pull request by jagerman**
This PR adds endpoints to storage server that allows someone to maintain a oxenmq connection to one or more swarm members and receive pushed notifications through that connection when new messages are delivered.

This is authenticated: subscribing requires a signature from the mailbox owner signed within the past 14 days, and connections have to be refreshed at least once/hour to keep the push notifications alive.

The immediate use for this will be for more efficient push notifications for mobile clients using the push notification server, but this mechanism will eventually also allow clients (over lokinet) to get messages pushed to them rather than having to frequently poll.

### CODE REVIEW

**Comment by ghost** [comment]
may I ask what this will be for regarding session?

**Comment by KeeJef** [comment]
Its an update for how we manage push notifications, basically refactoring Oxen storage server to notify the push notification server of any new messages for users who are subscribed to use push notifications. "Message monitoring" sounds scary but its really just monitoring to ensure PN's get set properly. Storage servers still cant read message contents or anything like that.

**Comment by majestrate** [comment]
On Mon, 25 Jul 2022 19:58:57 -0700
Flamez ***@***.***> wrote:

> may I ask what this will be for regarding session?
> 

this cuts the PN latency by 60% (it is 1/3 the rtt as a result)

**Comment by ghost** [comment]
> Its an update for how we manage push notifications, basically refactoring Oxen storage server to notify the push notification server of any new messages for users who are subscribed to use push notifications. "Message monitoring" sounds scary but its really just monitoring to ensure PN's get set properly. Storage servers still cant read message contents or anything like that.

Thanks for responding and yeah I did kind of think it was what the PR meant :P

**Comment by jagerman** [comment]
> Thanks for responding and yeah I did kind of think it was what the PR meant :P

I reworded the title to be less scary

File: oxenss/server/omq.h [hunk 1]
```diff
@@ -130,8 +130,8 @@ class OMQ {
     ///   - P -- an ed25519 pubkey underlying a session ID, in bytes (32 bytes).  The account
     ///     will be derived by converting to an x25519 pubkey and prepending the 0x05 byte.  The
     ///     signature uses *this* key, not the derived x25519 key.
-    /// - S -- (optional) a 32-byte authentication subkey to use for authentication.  The
-    ///   signature with such a subkey uses a derived key (as described in the RPC endpoint
+    /// - S -- (optional) a 32-byte authentication subkey tag to use for authentication.  The
```
Comment (majestrate) [inline oxenss/server/omq.h:133]: this sounds a bit more ambiguous as "tags" are also used in authenticated encryption as authentication tags.

Comment (jagerman) [reply oxenss/server/omq.h:133]: True, but the next line is "go see the rpc docs" which has a paragraph about the subkey value.

**Pull request merged**